query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -90,4 +101,37 @@ public static Expression createStringArrayLiteralValueExpression(List<String> va
                         .setValueType(ValueType.STRING_ARRAY)))
         .build();
   }
+
+  public static boolean isComplexAttributeExpression(Expression expression) {
+    return expression.getValueCase() == ValueCase.ATTRIBUTE_EXPRESSION
+        && expression.getAttributeExpression().hasSubpath();
+  }
+
+  public static boolean isColumnMapAttribute(Expression expression, ViewDefinition viewDefinition) {
+    return expression.getValueCase() == COLUMNIDENTIFIER
+        && isMapField(expression.getColumnIdentifier().getColumnName(), viewDefinition);
+  }
+
+  public static boolean isMapSubpathAttributeExpression(
+      Expression expression, ViewDefinition viewDefinition) {
+    return isComplexAttributeExpression(expression)
+        && isMapField(expression.getAttributeExpression().getAttributeId(), viewDefinition);
+  }
+
+  public static boolean isMapAttributeExpression(
+      Expression expression, ViewDefinition viewDefinition) {
+    return isColumnMapAttribute(expression, viewDefinition)
+        || isMapSubpathAttributeExpression(expression, viewDefinition);
```

Comment:
Isn't that the desired behavior?

Reply:
No. There should be zero difference in behavior between
```
      columnIdentifier {
        columnName: "EVENT.tags"
      }
```
and
```
      attributeExpression {
        key: "EVENT.tags"
      }
```

This was discussed in more detail up in https://github.com/hypertrace/query-service/pull/107#discussion_r764923949 , but I could see how it might be confusing.
> isMapAttributeExpression replaces everywhere we do isColumnMapAttribute(expression) || isAttributeExpressionMapAttribute(expression) - this both makes the later deprecation easier and fixes the bug discussed above

I meant that it should replace their usage, not that it should match their logic. This missing case is the one I was referencing.

Reply:
What kind of operation do we expect if such a request is given ?
```
   attributeExpression {
        key: "EVENT.tags"
      }
```

I think this is the case for `CONTAINS_KEY` and this will be added later.

Reply:
The main two usages of a map col without a subpath would be a simple selection or CONTAINS_KEY. This one line bug is I believe the only thing that's missing for CONTAINS_KEY support. 

I guess another improvement would be to add support for passing the arg as a string rather than a string array, which is a couple more lines and you could do in a separate PR if you wanted.

I get wanting to break things down - I'm the biggest proponent of that - but it doesn't make sense to do 99% of the work, then leaving it unusable until we put the last few lines of code in a separate PR.